lib/ex_unit/test/ex_unit_test.exs

```diff
@@ -32,6 +32,40 @@ defmodule ExUnitTest do
            end) =~ "\n0 failures\n"
   end
 
+  test "supports reruns" do
+    defmodule SampleAsyncTest do
+      use ExUnit.Case, async: true
+
+      test "true" do
+        assert false
+      end
+
+      test "false" do
+        assert false
+      end
+    end
+
+    configure_and_reload_on_exit([])
+
+    assert capture_io(fn ->
+             assert ExUnit.run() == %{
+                      failures: 2,
+                      skipped: 0,
+                      total: 2,
+                      excluded: 0
+                    }
+           end) =~ "\n2 tests, 2 failures\n"
+
+    assert capture_io(fn ->
+             assert ExUnit.rerun([SampleAsyncTest]) == %{
+                      failures: 2,
+                      skipped: 0,
+                      total: 2,
+                      excluded: 0
+                    }
+           end) =~ "\n2 tests, 2 failures\n"
```

Comment:
Great test! :+1:

Reply:
Thanks 😄 .

Just make `:ex_unit_async` as persisted attributes as suggested and it's now support adding the test modules correctly based on the attributes. 

But I'm not sure how should I test it as the result doesn't specified if an async test is run.